farm.js

```diff
@@ -3,23 +3,23 @@
 
 const Bird = require('bird-foo');
 const Cow = require('cow-foo');
-const feedCow = require('cow-foo/feed-cow');
 
 const feedBird = require('./lib/feed-bird');
+const feedCow = require('./lib/feed-cow');
 const eatLunch = require('./lib/eat-lunch');
 
-
 function main(cb) {
   const cow = new Cow();
-  feedCow(cow, function() {
-    var bird = new Bird();
-    var birdFood = (bird == 'pigeon') ? 'seed' : 'corn';
-    feedBird(bird, birdFood, function() {
-      eatLunch(function() {
-        cb();
-      });
-    });
-  });
+
+  const bird = new Bird();
+  const birdFood = (bird == 'pigeon') ? 'seed' : 'corn';
+
+  return Promise.all([
+      feedCow(cow),
+      feedBird(bird, birdFood)
+    ]).
+    then(()=> eatLunch()).
```

Comment:
To be sure that no arguments wll be passed to eatLunch. But here eatLunch can be passed direcrlt